src/lib/broadcast.jl

```diff
@@ -227,8 +227,8 @@ end
   out = dual_function(f).(args...)
   eltype(out) <: Dual || return (out, _ -> nothing)
   y = map(x -> x.value, out)
-  _back(ȳ, i) = unbroadcast(args[i], ((a, b) -> a*b.partials[i]).(ȳ, out))
-  back(ȳ) = ntuple(i -> _back(ȳ, i), N)
+  _back(ȳ, geti) = unbroadcast(geti(args), ((a, b) -> a * geti(b.partials)).(ȳ, out))
+  back(ȳ) = ntuple(i -> _back(ȳ, StaticGetter{i}()), N)
```

Comment:
Change for CuArrays, not sure this matters. CI doesn't seem to mind. Pushed to try out on another machine, a bit later.

Reply:
Some GPU times: no harm, but no help either. 
```
julia> xs = cu(randn(100_000));

julia> @btime CUDA.@sync Zygote.gradient(x -> sum(cbrt.(x)), $xs);
  119.805 μs (375 allocations: 10.56 KiB)
  119.872 μs (377 allocations: 10.61 KiB) # this PR 

julia> @btime CUDA.@sync Zygote.gradient(x -> sum(tanh.(x)), $xs);  # has its own gradient
  99.938 μs (342 allocations: 10.81 KiB)

julia> @btime CUDA.@sync Zygote.gradient(x -> sum((identity∘tanh).(x)), $xs);  # force generic version
  116.577 μs (372 allocations: 10.52 KiB)
  117.023 μs (371 allocations: 10.52 KiB) # this PR
```
Also, the generic is as fast as the specific. Remind me why we don't do this on the CPU, and have this 300x slower generic thing?

Reply:
I know nothing about how Zygote works with GPU, but taking a guess:
because moving memory between GPU./VRAM and CPU/RAM is even more expensive?

Reply:
I don't quite know why the generic broadcast isn't used for the GPU, perhaps something would break. It uses dual numbers instead, and perhaps this makes purity assumptions about the function involved?

If we very simply call that on the CPU, we get times like this: 
```
julia> @btime Zygote.gradient(x -> sum(abs.(x)), $xs);  # earlier 478.458 μs -> 28.625 μs
  15.209 μs (20 allocations: 313.16 KiB)

julia> @btime Zygote.gradient(x -> sum((identity∘tanh).(x)), $xs);  # earlier 29.680 ms
  125.166 μs (20 allocations: 313.16 KiB)
```
This is with https://github.com/FluxML/Zygote.jl/pull/1001/commits/922716865534a678631acf56bf680ab892f8eba5, which might be too crude... will think a bit & see what CI says.

Reply:
Here's what it gets wrong:
```
julia> gradient((x,y) -> sum((z->z^2+y[1]).(x)), [1,2,3], [4,5])
([2, 4, 6], nothing)  # this commit
([2, 4, 6], [3, 0])  # tagged Zygote
```

Reply:
> I don't quite know why the generic broadcast isn't used for the GPU, perhaps something would break. It uses dual numbers instead, and perhaps this makes purity assumptions about the function involved

I think the reason is that you can't send `IdDict`s to the GPU.
Which means you can't send `Zygote.Context` to the GPU.
Which means you can't run `Zygote._pullback`, which needs the `Context` to track some things, I am not sure what, but I think like tasks and globals?
but the things that the Context is mean to track don't need to be tracked in forwards mode.